app/controllers/wishlists_controller.rb

```diff
@@ -21,6 +21,11 @@ def create
     end
   end
 
+  def show
+    @wishlist = Wishlist.created_by(current_user).find_by_id(params[:id])
```

Comment:
Yeap. As @agalkiewicz suggested, catch the `RecordNotFound` error and act accordingly. What's more you don't need the `created_by` scope call here. It'll be verified by `authorize!` call anyways.